tests/test_packages/test_connections/test_p2p_libp2p/base.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Base classes for p2p_libp2p tests"""
+
+import atexit
+import logging
+import os
+from pathlib import Path
+from typing import List
+
+from aea.mail.base import Envelope
+from aea.multiplexer import Multiplexer
+
+from packages.fetchai.protocols.default.message import DefaultMessage
+
+from tests.conftest import TEMP_LIBP2P_TEST_DIR, remove_test_directory
+
+
+class BaseP2PLibp2pTest:
+    """Base class for p2p libp2p tests"""
+
+    cwd: str
+    t: str
+    tmp_dir: str
+    tmp_client_dir: str
+    log_files: List[str] = []
+    multiplexers: List[Multiplexer] = []
+    capture_log = True
+
+    @classmethod
+    def setup_class(cls):
+        """Set the test up"""
+
+        cls.cwd, cls.t = os.getcwd(), TEMP_LIBP2P_TEST_DIR
+        cls.tmp_dir = os.path.join(cls.t, "temp_dir")
+        cls.tmp_client_dir = os.path.join(cls.t, "tmp_client_dir")
+        Path(cls.tmp_dir).mkdir(parents=True, exist_ok=True)
+        os.chdir(cls.t)
+        atexit.register(cls.teardown_class)
+
+    @classmethod
+    def teardown_class(cls):
+        """Tear down the test"""
+
+        logging.debug(f"Cleaning up {cls.__name__}")
+        for mux in cls.multiplexers:
+            for con in mux.connections:
+                con.disconnect()
+            mux.disconnect()
+        cls.multiplexers.clear()
+        cls.log_files.clear()
+        os.chdir(cls.cwd)
+        if Path(cls.t).exists():
+            remove_test_directory(cls.t)
```

Comment:
only removing if not already done, prevents a lot of unnecessary logging at the end of tests regarding attempting to remove tmp directories already removed